Cargo.toml

```diff
@@ -20,6 +20,7 @@ license = "MIT/Apache-2.0"
 futures-core = "0.3.0"
 pin-project = "1.0.0"
 tokio = { version = "1.0", features = ["sync"] }
+static_assertions = "1.1.0"
```

Comment:
Ah, this should be moved to `dev-dependencies` since it's only used for testing. Just place the actual assertion behind a `#[cfg(test)]`.

Reply:
I was following the recommended [usage](https://docs.rs/static_assertions/1.1.0/static_assertions/#usage).

One of the benefits of keeping it in dependencies is that the assertion will fire during development, so if somebody causes the assertion to fail, tools like rust-analyzer will detect it. As a user of this crate, I would prefer that the crate would fail to compile and point out the assertion that failed than have to try to find out where the `Sync` went missing again 😅

Reply:
No, I'd prefer to not impose this added dependency on consumers, so let's move it to `dev-dependencies`. Your point about early development warnings is a good one, but at the same time, most editor integration these days check `cfg(test)` as well, so this will likely surface anyway :)